File: Shared/Blog.Contracts/V1/Responses/CommentsResponses/CommentResponse.cs
namespace Blog.Contracts.V1.Responses.CommentsResponses
{
    using System;
    using System.ComponentModel.DataAnnotations;
    using Blog.Contracts.V1.Responses.UsersResponses;

    /// <summary>
    /// Comment response.
    /// </summary>
    public class CommentResponse
    {
        /// <summary>
        /// Gets or sets the identifier.
        /// </summary>
        /// <value>
        /// The identifier.
        /// </value>
        public int? Id { get; set; }

        /// <summary>
        /// Gets or sets the post identifier.
        /// </summary>
        /// <value>
        /// The post identifier.
        /// </value>
        [Required]
        public int PostId { get; set; }

        /// <summary>
        /// Gets or sets the comment body.
        /// </summary>
        /// <value>
        /// The comment body.
        /// </value>
        [Required]
        public string CommentBody { get; set; }

        /// <summary>
        /// Gets or sets the created at.
        /// </summary>
        /// <value>
        /// The created at.
        /// </value>
        [DataType(DataType.DateTime)]
        public DateTime CreatedAt { get; set; }

        /// <summary>
        /// Gets or sets the likes.
        /// </summary>
        /// <value>
        /// The likes.
        /// </value>
        public int Likes { get; set; }

        /// <summary>
        /// Gets or sets the dislikes.
        /// </summary>
        /// <value>
        /// The dislikes.
        /// </value>
        public int Dislikes { get; set; }

        /// <summary>
        /// Gets or sets the user identifier.
        /// </summary>
        /// <value>
        /// The user identifier.
        /// </value>
        public string UserId { get; set; }

        /// <summary>
        /// Gets or sets the user.
        /// </summary>
        /// <value>
        /// The user.
        /// </value>
        public virtual ApplicationUserResponse User { get; set; }

        /// <summary>
        /// Gets or sets the name.
        /// </summary>
        /// <value>
        /// The name.
        /// </value>
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets the email.
        /// </summary>
        /// <value>
        /// The email.
        /// </value>
        public string Email { get; set; }
    }
}

File: Shared/Blog.FSharp.Contracts/V1/Responses/CommentsResponses/CommentResponse.fs
namespace Blog.FSharp.Contracts.V1.Responses.CommentsResponses

open System
open System.ComponentModel.DataAnnotations
open Blog.FSharp.Contracts.V1.Responses.UsersResponses

/// <summary>
/// Comment response.
/// </summary>
type CommentResponse() =
    /// <summary>
    /// Gets or sets the identifier.
    /// </summary>
    member val Id: int option = None with get, set

    /// <summary>
    /// Gets or sets the post identifier.
    /// </summary>
    [<Required>]
    member val PostId: int = 0 with get, set

    /// <summary>
    /// Gets or sets the comment body.
    /// </summary>
    [<Required>]
    member val CommentBody: string = String.Empty with get, set

    /// <summary>
    /// Gets or sets the created at.
    /// </summary>
    [<DataType(DataType.DateTime)>]
    member val CreatedAt: DateTime = DateTime.MinValue with get, set

    /// <summary>
    /// Gets or sets the likes.
    /// </summary>
    member val Likes: int = 0 with get, set

    /// <summary>
    /// Gets or sets the dislikes.
    /// </summary>
    member val Dislikes: int = 0 with get, set

    /// <summary>
    /// Gets or sets the user identifier.
    /// </summary>
    member val UserId: string = String.Empty with get, set

    /// <summary>
    /// Gets or sets the user.
    /// </summary>
    member val User: ApplicationUserResponse option = None with get, set

    /// <summary>
    /// Gets or sets the name.
    /// </summary>
    member val Name: string = String.Empty with get, set

    /// <summary>
    /// Gets or sets the email.
    /// </summary>
    member val Email: string = String.Empty with get, set

